src/XRP/model/xrp-account-set.ts

```diff
@@ -16,15 +17,56 @@ export default class XrpAccountSet {
    * @return an XrpAccountSet with its fields set via the analogous protobuf fields.
    * @see https://github.com/ripple/rippled/blob/3d86b49dae8173344b39deb75e53170a9b6c5284/src/ripple/proto/org/xrpl/rpc/v1/transaction.proto#L100
    */
-  public static from(accountSet: AccountSet): XrpAccountSet | undefined {
+  public static from(accountSet: AccountSet): XrpAccountSet {
     const clearFlag = accountSet.getClearFlag()?.getValue()
     const domain = accountSet.getDomain()?.getValue()
+    // domain must be lowercase
+    if (domain?.toLowerCase() != domain) {
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'AccountSet protobuf field `domain` is not lowercase.',
+      )
+    }
     const emailHash = accountSet.getEmailHash()?.getValue_asU8()
     const messageKey = accountSet.getMessageKey()?.getValue_asU8()
     const setFlag = accountSet.getSetFlag()?.getValue()
+    if (clearFlag && setFlag && clearFlag == setFlag) {
```

Comment:
can we use `===`? is type coercion here on purpose?

Reply:
No, I just forgot that Javascript uses `===`